cineast-core/src/main/java/org/vitrivr/cineast/core/features/VisualTextCoEmbedding.java

```diff
@@ -220,6 +254,144 @@ private float[] embedImage(BufferedImage image) {
     }
   }
 
+  /**
+   * Helper to convert images to frames
+   *
+   * @param images are converted to frames
+   * @return List<MultiImage> for video embedding
+   */
+  private List<MultiImage> framesFromImages(List<BufferedImage> images) {
+    var frames = new ArrayList<MultiImage>();
+    for (BufferedImage image : images) {
+      var factory = CachedDataFactory.getDefault();
+      frames.add(factory.newInMemoryMultiImage(image));
+    }
+    return frames;
+  }
+
+  /**
+   * This method takes a list of images and determines, based on {@link ViewpointStrategy}, the embed vector which describes the images most precise. This method can be simplified, once the best strategy is determined.
+   *
+   * @param images            the list of Images to embed
+   * @param viewpointStrategy the strategy to find the vector
+   * @return the embedding vector
+   */
+  private float[] embedMostRepresentativeImages(List<BufferedImage> images, ViewpointStrategy viewpointStrategy) {
```

Comment:
This looks like code that is not specific to this feature, so it might make sense to extract it to be usable for related purposes as well.